### DISCUSSION TITLE
Nginx configuration and setup

### DISCUSSION THREAD

**Discussion by srajasimman** [Sunbird-cQube/community#31] [Q&A] [answered]
Hi @jai-ekstep,

As part of cQube installation, we have followed the steps mentioned here [README.md](https://github.com/Sunbird-cQube/cQube_Base/blob/release-3.3/README.md)

Also configured Nginx to forward requests to the cQube backend server [config attached here](https://gist.github.com/srajasimman/1fe90debefaf14f6fc1b475ce4d272d4) 

We would like to understand Nginx config is correct.

we found that ports `4200` and `3000` are not listening after the successful installation.

Please do let us know any further setup is required to complete the setup.

cc: @gandham-santhosh

**Comment by SreenivasNimmagadda** [comment]
Hi Rajasimman,

From your configuration file, It seems all the details are correct. 
Both the ports 4200 & 3000 are expected to run properly. 

Please provide us few more below details to understand the issue.
1.  Error which you are getting in the browser while accessing the web page.
2. Please run the 'netstat' command in cQube server and send us the output. 

Thanks.

**Comment by srajasimman** [comment, marked as answer]
Hi @SreenivasNimmagadda 

We have completed the installation of [Sunbird-cQube/cQube_Base](https://github.com/Sunbird-cQube/cQube_Base)
expecting all ports mentioned the README.md should listen after.

Here are the details you asked for.

1. both the ports 4200 & 3000 are not even listening after the [Sunbird-cQube/cQube_Base](https://github.com/Sunbird-cQube/cQube_Base) installation.
2. `netstat` output
```
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 127.0.0.1:9990          0.0.0.0:*               LISTEN      29250/java          
tcp        0      0 127.0.0.1:41927         0.0.0.0:*               LISTEN      32575/java          
tcp        0      0 10.14.0.4:9000          0.0.0.0:*               LISTEN      31997/grafana-serve 
tcp        0      0 0.0.0.0:8080            0.0.0.0:*               LISTEN      29250/java          
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      1212/systemd-resolv 
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1827/sshd           
tcp        0      0 10.14.0.4:5432          0.0.0.0:*               LISTEN      28151/postgres      
tcp        0      0 127.0.0.1:5432          0.0.0.0:*               LISTEN      28151/postgres      
tcp        0      0 0.0.0.0:8443            0.0.0.0:*               LISTEN      29250/java          
tcp        0      0 0.0.0.0:8096            0.0.0.0:*               LISTEN      32575/java          
tcp        0      0 0.0.0.0:8000            0.0.0.0:*               LISTEN      28498/nginx: master 
tcp        0      0 0.0.0.0:8001            0.0.0.0:*               LISTEN      28498/nginx: master 
tcp6       0      0 :::9100                 :::*                    LISTEN      30145/node_exporter 
tcp6       0      0 127.0.0.1:38673         :::*                    LISTEN      32552/java          
tcp6       0      0 :::22                   :::*                    LISTEN      1827/sshd           
tcp6       0      0 :::9090                 :::*                    LISTEN      30059/prometheus
```

Thanks
Rajasimman

cc: @gandham-santhosh

**Reply by SreenivasNimmagadda** [reply]
Dear Rajasimman,

The ports 4200 & 3000 will be listening after the installation of [cQube_Workflow](https://github.com/Sunbird-cQube/cQube_Workflow). 

Thanks.

**Reply by srajasimman** [reply]
Hi @SreenivasNimmagadda 

Thanks for the reply. 
I will install [cQube_Workflow](https://github.com/Sunbird-cQube/cQube_Workflow) and you know if any.

Rajasimman S

**Reply by SreenivasNimmagadda** [reply]
Sure. Thanks
